src/string.rs

```diff
@@ -140,6 +190,43 @@ impl<const N: usize> String<N> {
         self.vec.extend_from_slice(string.as_bytes())
     }
 
+    /// Append a given UTF-16-encoded slice onto the end of this `String`.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// use heapless::String;
+    ///
+    /// // 𝄞music
+    /// let mut s: String<10> = String::new();
+    /// let v = &[0xD834, 0xDD1E, 0x006d, 0x0075,
+    ///           0x0073, 0x0069, 0x0063];
+    /// s.push_utf16(v).unwrap();
+    /// assert_eq!(s, "𝄞music");
+    ///
+    /// // 𝄞mu<invalid>ic
+    /// let mut s: String<10> = String::new();
+    /// let v = &[0xD834, 0xDD1E, 0x006d, 0x0075,
+    ///           0xD800, 0x0069, 0x0063];
+    /// assert!(s.push_utf16(v).is_err());
+    /// ```
+    pub fn push_utf16(&mut self, v: &[u16]) -> Result<(), Error> {
```

Comment:
this is not an API provided in the standard library. I'd like to keep the API surface of `heapless::String` as close as libstd's. could you please make this method private?

Reply:
I specifically need this actually for building a file path string while walking through a FAT32 volume.

Reply:
this is not a method that requires access to private parts of `String` so it can be implemented out of tree, either as a free function or as an extension method.

I'd still prefer to omit this. it'd be problematic if upstream libstd adds a method named the same that has a different signature or semantics, or add this functionality under a different name.

Reply:
> this is not a method that requires access to private parts of String so it can be implemented out of tree, either as a free function or as an extension method.

In that case I would have to essentially copy-paste this function, which seems unnecessary.

> if upstream libstd adds a method named the same that has a different signature or semantics

If this happens, you can bump the version of this crate. In the meantime, this provides useful functionality.

> or add this functionality under a different name

If this happens, worst case is the different name needs to be added and this one deprecated.

Reply:
I'd be fine if this was behind a feature, e.g. `non-libstd-methods`. I just don't want to reimplement the wheel and make this PR useless to me.

Reply:
> In that case I would have to essentially copy-paste this function, which seems unnecessary.

you would be maintaining code that you need; that seems normal to me. if you needed this extra API on `std::Vec` you would likely do the same.

> I'd be fine if this was behind a feature, e.g. non-libstd-methods

that would increase the maintenance work: the new Cargo feature would need to be tested in CI, in combination with other Cargo features. it also doesn't address the API issues; API behind a Cargo feature still need to adhere to semver.